curriculum/challenges/english/10-coding-interview-prep/algorithms/implement-binary-search.md

```diff
@@ -0,0 +1,177 @@
+1. If middle `value > target`, search left half of array in next compare.
+
+As you can see, you are successively halving an array, which gives your the log(n) efficiency. For this challenge, we want you to show your work - how you got to the target value... the path you took!
+
+# --instructions--
+
+Write a function `binarySearch` that implements the binary search algorithm on an array, returning the path you took (each middle value comparison) to find the target in an array.
+
+The function takes a sorted array of integers and a target value as input. It returns an array containing (in-order) the middle value you found at each halving of the original array until you found the target value. The target value should be the last element of the returned array. If value not is found, return the string `Value Not Found`.
+
+For example, `binarySearch([1,2,3,4,5,6,7], 5)` would return `[4,6,5]`.
+
+For this challenge, when halving, you MUST use `Math.floor()` when doing division: `Math.floor(x/2)`. This will give a consistent, testable path.
+
+**Note:** The following array will be used in tests:
+
+```js
+const testArray = [
+  0, 1, 2, 3, 4, 5, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22,
+  23, 49, 70
+];
+```
+
+# --hints--
+
+`binarySearch` should be a function.
+
+```js
+assert(typeof binarySearch == 'function');
+```
+
+`binarySearch(testArray, 0)` should return `[13, 5, 2, 0]`.
+
+```js
+assert.deepEqual(binarySearch(_testArray, 0), [13, 5, 2, 0]);
+```
+
+`binarySearch(testArray, 1)` should return [13, 5, 2, 0, 1].
+
+```js
+assert.deepEqual(binarySearch(_testArray, 1), [13, 5, 2, 0, 1]);
+```
+
+`binarySearch(testArray, 2)` should return [13, 5, 2].
+
+```js
+assert.deepEqual(binarySearch(_testArray, 2), [13, 5, 2]);
+```
+
+`binarySearch(testArray, 6)` should return 'Value Not Found'.
+
+```js
+assert.strictEqual(binarySearch(_testArray, 6), 'Value Not Found');
+```
+
+`binarySearch(testArray, 11)` should return [13, 5, 10, 11].
+
+```js
+
+let searchList = [
+  0, 1, 2, 3, 4, 5, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22,
+  23, 49, 70
+];
+
+let testArray = [13, 5, 10, 11];
+
+// user algo found value
+for (let j = 0; j < testArray.length; j++) {
+  if (binarySearch(searchList, 11)[j] != testArray[j]) {
+    
+    assert.fail();
+    break;
+  }
+}
+
+```
+
+`binarySearch(testArray, 13)` should return [13].
+
+```js
+let searchList = [
+  0, 1, 2, 3, 4, 5, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22,
+  23, 49, 70
+];
+
+let testArray = [13];
+
+// user algo found value
+for (let j = 0; j < testArray.length; j++) {
+  if (binarySearch(searchList, 13)[j] != testArray[j]) {
+    
+    assert.fail();
+    break;
+  }
+}
+
+```
+
+
+# --seed--
+
+## --after-user-code--
+
+## --seed-contents--
+
+```js
+function binarySearch(searchList, value) {
+  
+  let arrayPath = [];
+  // Only change code below this line
+
+  return arrayPath;
+  // Only change code above this line
```

Comment:
```suggestion
  return arrayPath;
```

the interview prep section doesn't need hand helding on where to edit code, so we can avoid comments that say where to edit